src/esm_runscripts/filedicts.py

```diff
@@ -73,6 +74,78 @@ def inner_method(self, *args, **kwargs):
     return inner_method
 
 
+def globbing(method):
+    """
+    Decorator method for ``SimulationFile``'s methods ``cp``, ``mv``, ``ln``, that
+    enables globbing. If a ``*`` is found on the ``source`` or ``target`` the globbing
+    logic is activated, and consist of:
+    - run checks for globbing syntax
+    - check if any file matches the globbing pattern
+    - construct one instance of ``SimulationFile`` for each file matching the globbing
+    - run the ``method`` for that particular file
+
+    Parameters
+    ----------
+    method : method
+        The decorated method (``cp``, ``mv``, ``ln``)
+
+    Returns
+    -------
+    method : method
+        If no globbing is needed, returns the method as it was given originally.
+    """
+    @functools.wraps(method)
+    def inner_method(self, source, target, *args, **kwargs):
+        method_name = method.__name__
+        source_name = self[f"name_in_{source}"]
+        target_name = self[f"name_in_{target}"]
+
+        if "*" in source_name or "*" in target_name:
+            # Get wildcard patterns
+            source_pattern = source_name.split("*")
+            target_pattern = target_name.split("*")
+
+            # Check wild cards syntax
+            self.wild_card_check(source_pattern, target_pattern)
+
+            # Obtain source files
+            glob_source_paths = self.find_globbing_files(source)
+
+            # Extract globbing source names
+            glob_source_names = []
+            for glob_source_path in glob_source_paths:
+                glob_source_names.append(pathlib.Path(glob_source_path).name)
```

Comment:
Could be simpler:
```suggestion
            # Extract globbing source names
            glob_source_names = [glob_source_path.name for glob_source_path in glob_source_paths]
```

Reply:
The problem here was that you need `pathlib.Path` because `glob` return a list of strings, not Paths.

Reply:
Then instead:
```suggestion
            # Extract globbing source names
            glob_source_names = [pathlib.Path(glob_source_path.name) for glob_source_path in glob_source_paths]
```